src/main/kotlin/io/github/anki/anki/controller/DecksController.kt

```diff
@@ -6,74 +6,81 @@ import io.github.anki.anki.controller.dto.PatchDeckRequest
 import io.github.anki.anki.controller.dto.mapper.toDeck
 import io.github.anki.anki.controller.dto.mapper.toDto
 import io.github.anki.anki.service.DeckService
+import io.github.anki.anki.service.model.Deck
 import io.github.anki.anki.service.secure.SecurityService
 import jakarta.validation.Valid
 import org.slf4j.LoggerFactory
-import org.springframework.http.HttpHeaders
 import org.springframework.http.HttpStatus
 import org.springframework.web.bind.annotation.DeleteMapping
 import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PatchMapping
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.PostMapping
 import org.springframework.web.bind.annotation.RequestBody
-import org.springframework.web.bind.annotation.RequestHeader
 import org.springframework.web.bind.annotation.RequestMapping
 import org.springframework.web.bind.annotation.ResponseStatus
 import org.springframework.web.bind.annotation.RestController
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
 
 @RestController
 @RequestMapping(DecksController.BASE_URL)
 class DecksController(
     private val service: DeckService,
-    val securityService: SecurityService,
+    private val securityService: SecurityService,
 ) {
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
     fun createDeck(
         @Valid @RequestBody request: NewDeckRequest,
-        @RequestHeader header: HttpHeaders,
-    ): DeckDtoResponse {
-        LOG.info("IN: $DecksController ${BASE_URL} with name ${request.name}")
-        val deck = service.createNewDeck(request.toDeck(securityService.jwtUtils.getUserIdFromAuthHeader(header)))
-        LOG.info("OUT: $DecksController ${BASE_URL} create deck with id = ${deck.id}")
-        return deck.toDto()
-    }
+    ): Mono<DeckDtoResponse> =
+        securityService
+            .getUserIdFromAuthentication()
+            .doFirst { LOG.info("IN: $DecksController $BASE_URL create deck with name {}", request.name) }
+            .flatMap {
+                service.createNewDeck(request.toDeck(it))
+            }
+            .map(Deck::toDto)
+            .doOnNext { LOG.info("OUT: $DecksController $BASE_URL created deck with id = {}", it.id) }
 
     @GetMapping
     @ResponseStatus(HttpStatus.OK)
-    fun getDecks(@RequestHeader header: HttpHeaders): List<DeckDtoResponse> {
-        LOG.info("IN: $DecksController ${BASE_URL} get decks")
-        val decks = service.getDecks(securityService.jwtUtils.getUserIdFromAuthHeader(header))
-        LOG.info("OUT: $DecksController ${BASE_URL} get all decks")
-        return decks.map { it.toDto() }
-    }
+    fun getDecks(): Flux<DeckDtoResponse> =
+        securityService
+            .getUserIdFromAuthentication()
+            .doFirst { LOG.info("IN: $DecksController $BASE_URL get decks") }
+            .flatMapMany { service.getDecks(it) }
+            .map(Deck::toDto)
```

Comment:
Why here you use the function reference but in the `flatMapMany` you use lambda?

Be consisntent